tests/helpers/test_entity_platform.py

```diff
@@ -558,24 +558,32 @@ async def test_async_remove_with_platform_update_finishes(hass: HomeAssistant) -
     component = EntityComponent(_LOGGER, DOMAIN, hass)
     await component.async_setup({})
     entity1 = MockEntity(name="test_1")
+    entity2 = MockEntity(name="test_1")
 
     async def _delayed_update(*args, **kwargs):
-        await asyncio.sleep(0.01)
+        update_called.set()
+        await update_done.wait()
 
     entity1.async_update = _delayed_update
-
-    # Add, remove, add, remove and make sure no updates
-    # cause the entity to reappear after removal
-    for _ in range(2):
-        await component.async_add_entities([entity1])
-        assert len(hass.states.async_entity_ids()) == 1
-        entity1.async_write_ha_state()
-        assert hass.states.get(entity1.entity_id) is not None
-        task = asyncio.create_task(entity1.async_update_ha_state(True))
-        await entity1.async_remove()
-        assert len(hass.states.async_entity_ids()) == 0
+    entity2.async_update = _delayed_update
+
+    # Add, remove, and make sure no updates
+    # cause the entity to reappear after removal and
+    # that we can add another entity with the same entity_id
+    for entity in [entity1, entity2]:
```

Comment:
With the this updated test, does the code in dev branch fail under test? Otherwise maybe we should add another test that targets the PR title?

Reply:
I added tests in `tests/helpers/test_entity.py` for the changed behavior, I must have forgotten to push them 🤦